packages/roosterjs-content-model-plugins/lib/autoFormat/link/createLinkAfterSpace.ts

```diff
@@ -15,22 +15,30 @@ export function createLinkAfterSpace(editor: IEditor) {
             const length = selectedSegmentsAndParagraphs[0][1].segments.length;
             const marker = selectedSegmentsAndParagraphs[0][1].segments[length - 1];
             const textSegment = selectedSegmentsAndParagraphs[0][1].segments[length - 2];
+
             if (
                 marker.segmentType == 'SelectionMarker' &&
                 textSegment.segmentType == 'Text' &&
                 !textSegment.link
             ) {
                 const link = textSegment.text.split(' ').pop();
-                if (link && matchLink(link)) {
+                const url = link?.trimRight();
+                if (url && link && matchLink(url)) {
                     textSegment.text = textSegment.text.replace(link, '');
-                    const linkSegment = createText(link, marker.format, {
+                    const linkSegment = createText(url, marker.format, {
                         format: {
-                            href: link,
+                            href: url,
                             underline: true,
                         },
                         dataset: {},
                     });
+
                     selectedSegmentsAndParagraphs[0][1].segments.splice(length - 1, 0, linkSegment);
```

Comment:
What if cursor is not at the end of paragraph? We should use index of current selection, but not the whole length